src/fsharp/PatternMatchCompilation.fsi

```diff
@@ -50,7 +51,10 @@ val ilFieldToTastConst: ILFieldInit -> Const
 val internal CompilePattern: 
     TcGlobals ->
     DisplayEnv ->
-    Import.ImportMap -> 
+    Import.ImportMap ->
+    // tcVal
+    (ValRef -> ValUseFlag -> TTypes -> range -> Expr * TType) ->
```

Comment:
Not super happy with pushing tcval like this into here but ordering issues forced my hand.

Reply:
This is generally fine, since it's all internal. We add parameters all the time to provide additional context for things like this. Might be good to just give it an explicit name in the annotation as well.

Reply:
This is ok, it's annoying but needed for late codegen of TAST fragments